Documentation/cross-machine-test-guide.md

```diff
@@ -0,0 +1,132 @@
+service(s) on a machine other than the machine running the test
+(example: cross-platform testing).
+
+For this we have developed what we call the "Bridge." This Bridge is
+a WebAPI application meant to run on a Windows OS using the full
+.NET framework.  It offers a REST API that the scenario tests can
+invoke through normal HTTP requests to launch WCF services on other
+machines.
+
+The Bridge itself is agnostic of WCF services. Instead, it is aware
+of types that implement IResource, and it uses reflection to analyze
+a collection of assemblies to discover them.  It then exposes each of these
+as a "resource" to the Bridge REST API.  When a test requests a particular
+resource, the Bridge invokes the appropriate IResource.  In this case, the
+WCF tests provide IResources that know how to create and host WCF services.
+
+When the tests start up, they inform the Bridge where to find these
+resource assemblies.  This folder is known as the "Bridge Resource Folder".
+
+There are several Bridge-specific properties that are meaningful to
+both the Bridge and the tests.  They can be defined either as Environment
+variables or passed into build.cmd as MSBuild properties.  They are:
+
+  - **BuildHost**: the name of the machine on which the Bridge runs (default localhost)
+  - **BridgePort**: the port to use to communicate with the Bridge (default 44283)
+  - **BridgeResourceFolder**: the full path to the folder that contains the Bridge IResource assemblies (default bin\Wcf\Bridge\Resources)
+  - **BridgeAllowRemote**: indicates whether the Bridge will accept requests from machines other than itself (default: false)
+
+Starting the Bridge Locally
+---------------------------
+By default, the Bridge will self-start locally during OuterLoop tests and accept only
+requests from the machine running the tests.  It will shutdown when all the OuterLoop
+tests have completed.
+
+If you want to run the Bridge on a different machine, you must follow the steps below.
+
+Starting the Bridge on a different machine
+------------------------------------------
+On the Bridge where the machine will run, do these things:
+
+cd to the root folder of a local WCF Git repository and run this command:
+
+```
+    startBridge.cmd {options}
+```
+
+The options can be any of these:
+
+    -portNumber NNN
+    -allowRemote true/false
+    -remoteAddresses comma-separated-list
+
+The 'portNumber' option allows you to choose a port other than
+the default 44283.
+
+The 'allowRemote' option tells the Bridge it is allowed to accept
+requests from other machines.  If unspecified, it will accept only
+from localhost.  It must be set to true if the Bridge is running
+on a machine other than where the tests run.
+
+The 'remoteAddresses' is a comma-separated list of IP addresses,
+a range of IP's, or one of the supported predefined terms supported
+for the Scope properties page in the Windows Firewall.
+See https://technet.microsoft.com/en-us/library/dd759059.aspx .
+
```

Comment:
This statement about 'remoteAddresses' is a partial lie.  I can set values like LocalSubnet using this mechanism, but I have not found a way in the API to set other predefined names like "intranet".  I will continue to investigate.